Source/WebCore/Modules/reporting/ReportingObserver.cpp

```diff
@@ -53,15 +77,52 @@ ReportingObserver::~ReportingObserver()
 
 void ReportingObserver::disconnect()
 {
+    m_reporting->unregisterReportingObserver(*this);
 }
 
 void ReportingObserver::observe()
 {
+    m_reporting->registerReportingObserver(*this);
+
+    if (!m_options.buffered)
+        return;
+
+    m_options.buffered = false;
+
+    // For each report in global’s report buffer, queue a task to execute § 4.3 Add report to observer with report and the context object.
+    m_reporting->appendQueuedReportForRelevantType(*this);
+}
+
+auto ReportingObserver::takeRecords() -> Vector<Ref<Report>>
+{
+    return WTFMove(m_queuedReports);
 }
 
-Vector<Report> ReportingObserver::takeRecords()
+void ReportingObserver::appendQueuedReportIfCorrectType(const Ref<Report>& report)
```

Comment:
Is this https://w3c.github.io/reporting/#add-report?
If so, dropping a URL would be helpful.

Reply:
Done